src/ui/display.c

```diff
@@ -318,15 +318,15 @@ bool prepare_display_output() {
 
     // token_index == 0 means HTR, else use token_index-1 as index on the tokens array
     if (token_index == 0) {
-        strcpy(symbol, "HTR");
+        strlcpy(symbol, "HTR", 4);
         symbol_len = 3;
     } else {
         // custom token
         token_symbol_t* token = G_context.tx_info.tokens[token_index - 1];
-        strcpy(symbol, token->symbol);
+        strlcpy(symbol, token->symbol, MAX_TOKEN_SYMBOL_LEN + 1);
```

Comment:
Why `MAX_TOKEN_SYMBOL_LEN + 1` and not `strlen(token->symbol) + 1`?

Reply:
The idea of strlcpy is to have an upper bound of memory to read, meaning that we allow the function to read up to N-1 chars (+NULL terminator) and on strings longer than the limit or non-null terminated strings it copies a valid string to the destination to avoid runtime errors.

The strlen has undefined behavior on non-null terminated strings since it counts up to the first 0x00 byte.
If we used the strlen we may add an invalid upper bound and defeat the purpose of using strlcpy.